### PR TITLE
Chore: Add request to verify conf files association

### PR DESCRIPTION
**Pull request by WilsonZiweiWang**
The `.conf` files should always be associated with `bitbake` language mode as long as there is no other extension causing the conflict. Currently, this extension is known to cause such conflict: [Txt Syntax](https://github.com/xshrim/vscode-txt-syntax/blob/3e5851b4a4eaf6a0e5fc5281fa29f1f68cd5d591/src/extension.js#L103)

I prefer to let users be aware of this situation rather than attempting to compete against another extension to set the language mode.

### CODE REVIEW

File: client/src/extension.ts [hunk 1]
```diff
@@ -52,10 +50,22 @@ export async function activate (context: ExtensionContext): Promise<void> {
   // Create the language client and start the client.
   client = new LanguageClient('bitbake', 'Language Server Bitbake', serverOptions, clientOptions)
 
+  client.onRequest('custom/verifyConfigurationFileAssociation', async (param) => {
```
Comment (deribaucourt) [inline client/src/extension.ts:53]: In my latest work, I tried to keep extension.ts as tidy as possible. It's now starting to have a lot of features so it needs structure IMO. I suggest to move this code in the src/language folder to not bloat the activate() function.

File: client/src/extension.ts [hunk 2]
```diff
@@ -52,10 +50,22 @@ export async function activate (context: ExtensionContext): Promise<void> {
   // Create the language client and start the client.
   client = new LanguageClient('bitbake', 'Language Server Bitbake', serverOptions, clientOptions)
 
+  client.onRequest('custom/verifyConfigurationFileAssociation', async (param) => {
+    if (param.filePath?.endsWith('.conf') === true) {
```
Comment (deribaucourt) [inline client/src/extension.ts:54]: There are .conf files that are not related to bitbake. Could we implement a more complete test that tries to guess if this is a bitbake .conf? (some paths are obvious, or some syntax elements?)

File: client/src/extension.ts (hunk 1 above)
Comment (WilsonZiweiWang) [reply client/src/extension.ts:53]: sure, I just know your PR was merged. I can rebase now

**Review by deribaucourt — APPROVED** [review]

File: client/src/extension.ts [hunk 3]
```diff
@@ -52,10 +50,22 @@ export async function activate (context: ExtensionContext): Promise<void> {
   // Create the language client and start the client.
   client = new LanguageClient('bitbake', 'Language Server Bitbake', serverOptions, clientOptions)
 
+  client.onRequest('custom/verifyConfigurationFileAssociation', async (param) => {
+    if (param.filePath?.endsWith('.conf') === true) {
+      const doc = await workspace.openTextDocument(param.filePath)
+      const { languageId } = doc
+      //  The modifications from other extensions may happen later than this handler, hence the setTimtOut
+      setTimeout(() => {
+        if (languageId !== 'bitbake') {
```
[the remark this answers is not in the document]
Comment (WilsonZiweiWang) [reply client/src/extension.ts]: Added current languageId in the message
